ethers-solc/src/project_util.rs

```diff
@@ -0,0 +1,191 @@
+        &self.project().paths
+    }
+
+    /// The root path of the temporary workspace
+    pub fn root(&self) -> &Path {
+        self.project().paths.root.as_path()
+    }
+
+    /// Returns the handle to the tempdir and the project
+    ///
+    /// NOTE: the `TempDir` object deletes its directory on drop, also removing all the project's
+    /// content
+    pub fn split(self) -> (Project<T>, TempDir) {
+        (self.inner, self.root)
+    }
+
+    /// Copies a single file into the given dir
+    pub fn copy_file(&self, source: impl AsRef<Path>, target_dir: impl AsRef<Path>) -> Result<()> {
+        let source = source.as_ref();
+        let target = target_dir.as_ref().join(
+            source
+                .file_name()
+                .ok_or_else(|| SolcError::msg(format!("No file name for {}", source.display())))?,
+        );
+
+        fs_extra::file::copy(source, target, &file_copy_options())?;
+        Ok(())
+    }
+
+    /// Copies all content of the source dir into the target dir
+    pub fn copy_dir(&self, source: impl AsRef<Path>, target_dir: impl AsRef<Path>) -> Result<()> {
+        fs_extra::dir::copy(source, target_dir, &dir_copy_options())?;
+        Ok(())
+    }
+
+    /// Copies a single file into the projects source
+    pub fn copy_source(&self, source: impl AsRef<Path>) -> Result<()> {
+        self.copy_file(source, &self.paths().sources)
+    }
+
+    pub fn copy_sources<I, S>(&self, sources: I) -> Result<()>
+    where
+        I: IntoIterator<Item = S>,
+        S: AsRef<Path>,
+    {
+        for path in sources {
+            self.copy_source(path)?;
+        }
+        Ok(())
+    }
+
+    /// Copies a single file into the project's main library directory
+    pub fn copy_lib(&self, lib: impl AsRef<Path>) -> Result<()> {
+        let lib_dir = self
+            .paths()
+            .libraries
+            .get(0)
+            .ok_or_else(|| SolcError::msg("No libraries folders configured"))?;
+        self.copy_file(lib, lib_dir)
+    }
+
+    /// Copy a series of files into the main library dir
+    pub fn copy_libs<I, S>(&self, libs: I) -> Result<()>
+    where
+        I: IntoIterator<Item = S>,
+        S: AsRef<Path>,
+    {
+        for path in libs {
+            self.copy_lib(path)?;
+        }
+        Ok(())
+    }
+}
+
+impl TempProject<HardhatArtifacts> {
+    /// Creates an empty new hardhat style workspace in a new temporary dir
+    pub fn hardhat() -> Result<Self> {
+        let tmp_dir = TempDir::new("tmp_hh").map_err(|err| SolcError::io(err, "tmp_hh"))?;
+        let root = tmp_dir.path().to_path_buf();
+        let cache = tmp_dir.path().join("cache");
+        let cache = cache.join(SOLIDITY_FILES_CACHE_FILENAME);
+
+        let paths = ProjectPathsConfig::builder()
+            .cache(cache)
+            .sources(root.join("contracts"))
+            .artifacts(root.join("artifacts"))
+            .lib(root.join("node_modules"))
+            .root(root)
+            .build()?;
```

Comment:
Can't you use [`ProjectPathsConfig::hardhat`](https://github.com/gakonst/ethers-rs/blob/5dec757493f15be016074e9355f7154d8e84aaae/ethers-solc/src/config.rs#L43) instead?

Reply:
nice, this simplifies it quite a bit